### PR TITLE
feat(oxa-writer): add basic oxa.dev JSON writer

### LINKED ISSUE #180: Basic oxa.dev JSON writer
### Component

Other

### Problem Statement

.

### Proposed Solution

.

### Alternatives Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by StephanMeijer**
## Summary

Adds a new `docspec-oxa-writer` crate that streams DocSpec events into [oxa.dev](https://oxa.dev/) JSON. KISS scope: only `Document`, `Paragraph`, and `Text` are emitted. All other events are silently dropped.

Closes #180

## Scope

This PR adds **only** the standalone `docspec-oxa-writer` crate. The `docspec` facade integration (`OutputFormat::Oxa`, `AnyWriter::Oxa`, `oxa` feature flag, `.oxa` extension detection) is **deferred** until `docspec-oxa-writer` has a published version on crates.io.

### New crate: `crates/docspec-oxa-writer/`
- `OxaWriter<W: Write>` implementing `EventSink`
- Streaming JSON emission via `docspec-json` (`JsonEmitter` + `StrusonBackend`) — zero buffering
- 15 unit tests covering all edge cases (empty document, orphan text, double StartDocument, JSON special chars, Unicode, etc.)

### Workspace
- Registered as workspace member in root `Cargo.toml`

## OXA JSON shape

```json
{"type":"Document","children":[
  {"type":"Paragraph","children":[
    {"type":"Text","value":"Hello, world"}
  ]}
]}
```

## Usage

```toml
[dependencies]
docspec-oxa-writer = { path = "..." }
```

```rust
use docspec_oxa_writer::OxaWriter;
use docspec_core::{Event, EventSink};

let mut buf = Vec::new();
let mut writer = OxaWriter::new(&mut buf);
// feed events...
writer.finish()?;
```

## Checklist
- [x] `cargo build -p docspec-oxa-writer` exits 0
- [x] `cargo test -p docspec-oxa-writer` — 15 passed, 0 failed
- [x] `cargo clippy -p docspec-oxa-writer --all-targets -- -D warnings` exits 0
- [x] `cargo build --workspace` exits 0
- [x] All public items have doc comments

## Summary by CodeRabbit

* **New Features**
  * Introduced a streaming converter for document events to oxa.dev JSON format with minimal memory overhead.

* **Chores**
  * Added new crate to workspace and updated release tooling configuration.

### CODE REVIEW

File: crates/docspec-oxa-writer/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+license.workspace = true
+repository.workspace = true
+readme = "README.md"
+description = "DocSpec event stream to oxa.dev JSON writer"
+keywords = ["document", "conversion", "streaming"]
+categories = ["encoding"]
+
+[dependencies]
+docspec-core = { path = "../docspec-core" }
+docspec-json = { path = "../docspec-json" }
```
Comment (StephanMeijer) [inline crates/docspec-oxa-writer/Cargo.toml]: Versions required here.

File: crates/docspec-oxa-writer/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            }
+            Event::EndDocument => {
+                if !self.document_open {
+                    return Ok(());
+                }
+                if self.in_paragraph {
+                    self.json.close_array()?;
+                    self.json.close_object()?;
+                    self.in_paragraph = false;
+                }
```
Comment (StephanMeijer) [inline crates/docspec-oxa-writer/src/lib.rs]: Duplicated also in `Event::EndParagraph`. Put in new function: `end_paragraph()` (or similar name)

File: crates/docspec-oxa-writer/Cargo.toml (hunk 1 above)
Comment (StephanMeijer) [reply crates/docspec-oxa-writer/Cargo.toml]: Addressed in 139dad5: both path dependencies now declare explicit versions (`docspec-core = "1.0.0"`, `docspec-json = "1.0.2"`), matching the convention used by `docspec-blocknote-writer`.

File: crates/docspec-oxa-writer/src/lib.rs (hunk 2 above)
Comment (StephanMeijer) [reply crates/docspec-oxa-writer/src/lib.rs]: Addressed in 139dad5: extracted a private `close_paragraph()` helper that owns the `if self.in_paragraph` guard, so both `EndDocument` and `EndParagraph` arms now share the same close logic. `EndParagraph` becomes a one-liner: `Event::EndParagraph => self.close_paragraph()`.

File: crates/docspec-oxa-writer/tests/writer.rs [hunk 3]
```diff
@@ -0,0 +1,284 @@
+#![allow(
+    clippy::expect_used,
+    clippy::indexing_slicing,
+    clippy::non_ascii_literal,
+    clippy::panic,
+    clippy::print_stdout,
+    clippy::tests_outside_test_module,
+    clippy::unwrap_in_result,
+    clippy::unwrap_used
+)]
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec-oxa-writer/tests/writer.rs]: Addressed in 139dad5: trimmed the crate-level allow-list to `#![allow(clippy::expect_used, clippy::unwrap_used)]` (matching the workspace-blessed pattern in `docspec-blocknote-writer/tests/writer.rs`), wrapped all tests in `#[cfg(test)] mod tests { ... }`, removed the debug `println!`, eliminated the `panic!` branch via the exact-equality refactor in thread #4, and refactored `try_run` to propagate errors with `?` instead of `.expect()`. The unicode test keeps a focused `#[allow(clippy::non_ascii_literal)]` with a `# Reason:` comment per workspace convention.

File: crates/docspec-oxa-writer/tests/writer.rs [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    let result = try_run(vec![end_doc()]);
+    match result {
+        Err(e) => {
+            let msg = e.to_string();
+            assert!(
+                msg.contains("open containers remain or no root value written"),
+                "unexpected error: {msg}"
+            );
+        }
+        Ok(_) => panic!("expected Err but got Ok"),
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec-oxa-writer/tests/writer.rs]: Addressed in 139dad5: replaced `msg.contains(...)` with an exact-equality assertion against the full `Display` output:

```rust
let err = try_run(vec![end_doc()]).expect_err("expected Err");
assert_eq!(
    err.to_string(),
    "JSON error: cannot finish: open containers remain or no root value written"
);
```

This also removes the `panic!("expected Err but got Ok")` branch in one shot.

File: crates/docspec/Cargo.toml [hunk 5]
```diff
@@ -19,20 +19,22 @@ html = ["dep:docspec-html-reader"]
...
-all-writers = ["blocknote"]
+all-writers = ["blocknote", "oxa"]
 full = ["all-readers", "all-writers", "json"]
 
 [dependencies]
 docspec-core = { path = "../docspec-core", version = "1.0.0" }
 docspec-json = { path = "../docspec-json", version = "1.0.2", optional = true }
 docspec-markdown-reader = { path = "../docspec-markdown-reader", version = "1.0.2", optional = true }
 docspec-blocknote-writer = { path = "../docspec-blocknote-writer", version = "1.0.2", optional = true }
+docspec-oxa-writer = { path = "../docspec-oxa-writer", optional = true }
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec/Cargo.toml]: Out of scope: `docspec` facade integration was removed from this PR (it will be added in a follow-up once `docspec-oxa-writer` is published). See updated PR description.

File: crates/docspec/tests/factory_oxa.rs [hunk 6]
```diff
@@ -0,0 +1,109 @@
+//! End-to-end integration tests for Markdown → OXA pipeline.
+
+#![allow(clippy::unwrap_used, clippy::expect_used, clippy::indexing_slicing)]
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec/tests/factory_oxa.rs]: Out of scope: `docspec` facade integration was removed from this PR (it will be added in a follow-up once `docspec-oxa-writer` is published). See updated PR description.

File: crates/docspec/tests/format.rs [hunk 7]
```diff
@@ -62,6 +62,13 @@ mod tests {
         assert_eq!(result, Some(docspec::OutputFormat::Blocknote));
     }
 
+    #[cfg(feature = "oxa")]
+    #[test]
+    fn detect_oxa_from_oxa() {
+        let result = docspec::detect_output_format(Path::new("file.oxa"));
+        assert_eq!(result, Some(docspec::OutputFormat::Oxa));
+    }
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec/tests/format.rs]: Out of scope: `docspec` facade integration was removed from this PR (it will be added in a follow-up once `docspec-oxa-writer` is published). See updated PR description.

File: crates/docspec/Cargo.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec/Cargo.toml]: No, out of scope.

File: crates/docspec-oxa-writer/tests/writer.rs [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+            start_para(),
+            start_para(),
+            text("a"),
+            end_para(),
+            end_para(),
+            end_doc(),
+        ]);
+        assert_eq!(
+            json,
+            r#"{"type":"Document","children":[{"type":"Paragraph","children":[{"type":"Text","value":"a"}]}]}"#
```
Comment (StephanMeijer) [inline crates/docspec-oxa-writer/tests/writer.rs]: would be way more readable if this weren't a string

File: crates/docspec-oxa-writer/src/lib.rs [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+            | Event::EndTableCell
+            | Event::EndTableHeader
+            | Event::EndTableRow
+            | Event::EndUnorderedListItem
+            | Event::FootnoteRef { .. }
+            | Event::Image { .. }
+            | Event::LineBreak
+            | Event::SoftBreak
+            | Event::StartBlockQuote { .. }
+            | Event::StartCaption { .. }
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec-oxa-writer/src/lib.rs:158]: This is by design for the PoC scope: this writer only knows about `Document`, `Paragraph`, and `Text`. Tables (and every other block container) are not supported at all, so when a paragraph appears inside an unsupported container, the intended behavior is to flatten its content up to the document root rather than swallow it. Tracking `skip_depth` would drop user content that this writer is capable of representing.

File: crates/docspec-oxa-writer/tests/writer.rs (hunk 8 above)
Comment (StephanMeijer) [reply crates/docspec-oxa-writer/tests/writer.rs]: Addressed in 2883f60: added `serde_json` as a dev-dep (mirroring `docspec-blocknote-writer/Cargo.toml`) and rewrote 12 structural assertions to use `serde_json::json!()`. The 2 escape-verification tests (`text_with_json_special_chars`, `text_with_unicode`) stay as raw-string comparisons because parsing into `Value` loses the escape detail they're meant to verify. The flagged `nested_paragraph_no_op` test now reads as:

```rust
assert_eq!(v, json!({
    "type": "Document",
    "children": [{
        "type": "Paragraph",
        "children": [{"type": "Text", "value": "a"}]
    }]
}));
```

File: crates/docspec-html-writer/src/lib.rs [hunk 10]
```diff
@@ -0,0 +1,336 @@
...
+        Ok(())
+    }
+}
+
+impl<W: Write> EventSink for HtmlWriter<W> {
+    #[inline]
+    fn finish(mut self) -> Result<()> {
+        self.serializer.writer.flush()?;
+        Ok(())
+    }
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec-html-writer/src/lib.rs:57]: Out of scope for this PR. This PR only adds `docspec-oxa-writer`; `docspec-html-writer` was scaffolded in a separate commit on `main` and only appeared in this PR's diff because GitHub computed it against a stale base. After rebasing (24f525f) onto the current main, html-writer is no longer in this PR's net diff.

**Comment by StephanMeijer** [comment]
@kody review
